### PR TITLE
Releases

### PR DESCRIPTION
**Pull request by TheMcMurder**
Lots of initial work. Nothing is real yet and there are still some CSS things to fix

New Table
Screenshot from 2019-06-03 16-17-55
Maybe allow expanding rows
Screenshot from 2019-06-03 16-18-23
Create/edit modal
Screenshot from 2019-06-03 16-18-03
Preview pane in modal
Screenshot from 2019-06-03 16-18-10

### CODE REVIEW

File: tmeister/static/src/common/error-boundary.component.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  render() {
+    if (this.state.hasError) {
+      // You can render any custom fallback UI
+      return <h1>Something went wrong.</h1>;
+    }
+
+    return this.props.children; 
+  }
+}
```
Comment (TheMcMurder) [inline tmeister/static/src/common/error-boundary.component.js:27]: Basic Error Boundary

File: tmeister/static/src/common/modal/modal.component.js [hunk 2]
```diff
@@ -3,39 +3,64 @@ import { makeStyles } from '@material-ui/core/styles'
...
+          <div className={classes.modalContents}>
+            {children}
+          </div>
+          {
+            bottomRowContents && (
+              <div className={classes.bottomRow}>
+                {bottomRowContents}
+              </div>
+            )
+          }
```
Comment (TheMcMurder) [inline tmeister/static/src/common/modal/modal.component.js:61]: Turning this very open modal into a "scroll-dal" A modal with a static header/footer and scrollable contents

File: tmeister/static/src/menu/main-menu.component.js [hunk 3]
```diff
@@ -53,6 +55,18 @@ export default function MainMenu(props) {
...
+          <Tooltip title="Release Notes">
+            <a
+              className={`${classes.menuItem} ${activeIcon("release-notes")}`}
+              href="/#/release-notes"
+            >
+              <img src='/resources/audit-icon.png' className={classes.icon} />
+            </a>
+          </Tooltip>
+        )
+      }
```
Comment (TheMcMurder) [inline tmeister/static/src/menu/main-menu.component.js:69]: "Feature Toggled off"

File: tmeister/static/src/release-notes/release-notes.helper.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+import markdownIt from 'markdown-it'
+export const md = new markdownIt()
```
Comment (TheMcMurder) [inline tmeister/static/src/release-notes/release-notes.helper.js:2]: configured markdownIt

File: tmeister/static/src/release-notes/release-notes.hooks.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  return [
+    loadingNotes,
+    notes,
+    () => setCount(count + 1)
+  ]
+
+}
+
+
+function mockResults() {
```
Comment (TheMcMurder) [inline tmeister/static/src/release-notes/release-notes.hooks.js:32]: mock data from here to the bottom

**Comment by internettrans** [comment]
oo cool -- I didn't review this code but just lurking on the screenshots and they look exciting.

File: tmeister/static/src/release-notes/create-edit/create-edit-form.component.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          value={body}
+          onChange={(e) => setBody(e.target.value)}
+        />
+      </div>
+    </form>
+  )
+
+  function handleSubmit(e) {
+    e.preventDefault()
+    console.log('submit')
```
Comment (blittle) [inline tmeister/static/src/release-notes/create-edit/create-edit-form.component.js:48]: I assume this has yet to be implemented

File: tmeister/static/src/release-notes/release-notes-table/release-note-row.component.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+  function previewBody (markdown = '') {
+    const split = markdown.split('\n')
+    let preview = markdown
+    if (inlinePreview && split.length > 5) {
+      const slice = split.slice(0, 5)
+      slice.push('...')
+      preview = slice.join('\n')
+    }
+    const html = md.render(preview) || '<div>No preview</div>'
```
Comment (blittle) [inline tmeister/static/src/release-notes/release-notes-table/release-note-row.component.js]: If preview is an empty string or `null` does `md.render` also return a false-y value?

File: tmeister/static/src/release-notes/release-note-preview/release-note-preview.component.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React, { useMemo } from 'react'
+import { md } from '../release-notes.helper.js'
+
+export default function ReleaseNotePreview (props) {
+  const { title, body } = props
+  const html = useMemo(() => md.render(body), [body])
+  return (
+    <div>
+      <h1>{title}</h1>
+      <div dangerouslySetInnerHTML={{__html: html}} />
```
Comment (blittle) [inline tmeister/static/src/release-notes/release-note-preview/release-note-preview.component.js:10]: Would you want a default if the preview is empty?

File: tmeister/static/src/release-notes/create-edit/create-edit-form.component.js (hunk 6 above)
Comment (TheMcMurder) [reply tmeister/static/src/release-notes/create-edit/create-edit-form.component.js:48]: Yes, This was getting big so I wanted to stop now and submit what I had.

File: tmeister/static/src/release-notes/release-notes-table/release-note-row.component.js (hunk 7 above)
Comment (TheMcMurder) [reply tmeister/static/src/release-notes/release-notes-table/release-note-row.component.js]: It does. For now I'm going to remove the `no preview` possibility and I'll tackle empty states here and below soon.

**Review by blittle — APPROVED** [review]

**Pull request merged**